cfgmassif.go

```diff
@@ -24,14 +24,29 @@ func cfgMassifReader(cmd *CmdCtx, cCtx *cli.Context) error {
 		}
 	}
 
-	if cmd.reader == nil {
-		if err = cfgReader(cmd, cCtx); err != nil {
+	logdir := cCtx.String("logdir")
+	logfile := cCtx.String("logfile")
+
+	if logdir != "" || logfile != "" {
+		opener := cfgOpener()
+		// if we want to read from stdin we configure stdin opener
+		if logfile == "-" {
+			cmd.log.Infof("configuring for stdin")
+			opener = cfgStdinOpener()
+		}
+		mr, err := NewLocalMassifReader(logger.Sugar, opener, cfgDirLister(), logdir, logfile)
+		if err != nil {
+			return err
+		}
+		cmd.massifReader = mr
+	} else {
+		reader, err := cfgReader(cmd, cCtx)
+		if err != nil {
 			return err
 		}
+		mr := massifs.NewMassifReader(logger.Sugar, reader, massifs.WithoutGetRootSupport())
```

Comment:
It is probably a bug that we had WithoutGetRootSupport set, that means the peak stack is not mapped in the context and so attempts to refer to the peaks needed from the earlier massifs will fail